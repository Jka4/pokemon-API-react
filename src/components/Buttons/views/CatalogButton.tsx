/* eslint-disable import/no-anonymous-default-export */
import React from 'react';
import { NavLink } from 'react-router-dom';

import { Button } from '@material-ui/core';
import styled from 'styled-components';
import { pageRoutes } from 'routes';

const CatalogButton: React.FC = () => {
  return (
    <NavLink to={pageRoutes.catalog}>
      <ButtonStyled variant="contained" color="secondary">
        SHOW ALL POKEMON
      </ButtonStyled>
    </NavLink>
  );
};

const ButtonStyled = styled(Button)`
  margin: 5px 5px 5px 10px;
  font-size: 16px;
`;

export default CatalogButton;
