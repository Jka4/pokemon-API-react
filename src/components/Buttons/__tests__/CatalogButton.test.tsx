import CatalogButton from '../views/CatalogButton';

import { mount } from 'enzyme';

import { BrowserRouter } from 'react-router-dom';

describe('<CatalogButton/> ', () => {
  const container = mount(
    <BrowserRouter basename="/">
      <CatalogButton />
    </BrowserRouter>,
  );

  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });
});
