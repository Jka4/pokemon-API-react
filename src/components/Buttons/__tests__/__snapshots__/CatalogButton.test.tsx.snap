// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogButton/>  renders correctly 1`] = `
<BrowserRouter
  basename="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CatalogButton>
      <NavLink
        to="/catalog/big/"
      >
        <Link
          aria-current={null}
          to={
            Object {
              "hash": "",
              "pathname": "/catalog/big/",
              "search": "",
              "state": null,
            }
          }
        >
          <LinkAnchor
            aria-current={null}
            href="/catalog/big/"
            navigate={[Function]}
          >
            <a
              aria-current={null}
              href="/catalog/big/"
              onClick={[Function]}
            >
              <Styled(WithStyles(ForwardRef(Button)))
                color="secondary"
                variant="contained"
              >
                <WithStyles(ForwardRef(Button))
                  className="sc-bdfBwQ futGBs"
                  color="secondary"
                  variant="contained"
                >
                  <ForwardRef(Button)
                    className="sc-bdfBwQ futGBs"
                    classes={
                      Object {
                        "colorInherit": "MuiButton-colorInherit",
                        "contained": "MuiButton-contained",
                        "containedPrimary": "MuiButton-containedPrimary",
                        "containedSecondary": "MuiButton-containedSecondary",
                        "containedSizeLarge": "MuiButton-containedSizeLarge",
                        "containedSizeSmall": "MuiButton-containedSizeSmall",
                        "disableElevation": "MuiButton-disableElevation",
                        "disabled": "Mui-disabled",
                        "endIcon": "MuiButton-endIcon",
                        "focusVisible": "Mui-focusVisible",
                        "fullWidth": "MuiButton-fullWidth",
                        "iconSizeLarge": "MuiButton-iconSizeLarge",
                        "iconSizeMedium": "MuiButton-iconSizeMedium",
                        "iconSizeSmall": "MuiButton-iconSizeSmall",
                        "label": "MuiButton-label",
                        "outlined": "MuiButton-outlined",
                        "outlinedPrimary": "MuiButton-outlinedPrimary",
                        "outlinedSecondary": "MuiButton-outlinedSecondary",
                        "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                        "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                        "root": "MuiButton-root",
                        "sizeLarge": "MuiButton-sizeLarge",
                        "sizeSmall": "MuiButton-sizeSmall",
                        "startIcon": "MuiButton-startIcon",
                        "text": "MuiButton-text",
                        "textPrimary": "MuiButton-textPrimary",
                        "textSecondary": "MuiButton-textSecondary",
                        "textSizeLarge": "MuiButton-textSizeLarge",
                        "textSizeSmall": "MuiButton-textSizeSmall",
                      }
                    }
                    color="secondary"
                    variant="contained"
                  >
                    <WithStyles(ForwardRef(ButtonBase))
                      className="MuiButton-root MuiButton-contained sc-bdfBwQ futGBs MuiButton-containedSecondary"
                      component="button"
                      disabled={false}
                      focusRipple={true}
                      focusVisibleClassName="Mui-focusVisible"
                      type="button"
                    >
                      <ForwardRef(ButtonBase)
                        className="MuiButton-root MuiButton-contained sc-bdfBwQ futGBs MuiButton-containedSecondary"
                        classes={
                          Object {
                            "disabled": "Mui-disabled",
                            "focusVisible": "Mui-focusVisible",
                            "root": "MuiButtonBase-root",
                          }
                        }
                        component="button"
                        disabled={false}
                        focusRipple={true}
                        focusVisibleClassName="Mui-focusVisible"
                        type="button"
                      >
                        <button
                          className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdfBwQ futGBs MuiButton-containedSecondary"
                          disabled={false}
                          onBlur={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          type="button"
                        >
                          <span
                            className="MuiButton-label"
                          >
                            SHOW ALL POKEMON
                          </span>
                          <WithStyles(memo)
                            center={false}
                          >
                            <ForwardRef(TouchRipple)
                              center={false}
                              classes={
                                Object {
                                  "child": "MuiTouchRipple-child",
                                  "childLeaving": "MuiTouchRipple-childLeaving",
                                  "childPulsate": "MuiTouchRipple-childPulsate",
                                  "ripple": "MuiTouchRipple-ripple",
                                  "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                  "rippleVisible": "MuiTouchRipple-rippleVisible",
                                  "root": "MuiTouchRipple-root",
                                }
                              }
                            >
                              <span
                                className="MuiTouchRipple-root"
                              >
                                <TransitionGroup
                                  childFactory={[Function]}
                                  component={null}
                                  exit={true}
                                />
                              </span>
                            </ForwardRef(TouchRipple)>
                          </WithStyles(memo)>
                        </button>
                      </ForwardRef(ButtonBase)>
                    </WithStyles(ForwardRef(ButtonBase))>
                  </ForwardRef(Button)>
                </WithStyles(ForwardRef(Button))>
              </Styled(WithStyles(ForwardRef(Button)))>
            </a>
          </LinkAnchor>
        </Link>
      </NavLink>
    </CatalogButton>
  </Router>
</BrowserRouter>
`;
